File: extract_for_review.js
const fs = require('fs');
const path = require('path');
const { execSync } = require('child_process');

const WORKSPACE = path.resolve('f:\\PROJECT\\DISGUISE-ID\\fullstack-disguise');
const STAGING = path.join(WORKSPACE, 'staging_extract');
const ZIP_NAME = path.join(WORKSPACE, 'disguise-ai-integration-source.zip');

function redactContent(content) {
    // JWT Secrets
    content = content.replace(/(JWT_SECRET=)[\w\-\.]+/g, '$1[REDACTED]');
    content = content.replace(/(jwt\.verify\(.*?, process\.env\.JWT_SECRET \|\| )['"].*?['"]/g, '$1"[REDACTED]"');
    
    // API Keys
    content = content.replace(/(x-api-key\s*:\s*)[\w\-]+/ig, '$1[REDACTED]');
    content = content.replace(/(IOT_API_KEY\s*=\s*)['"].*?['"]/g, '$1"[REDACTED]"');
    content = content.replace(/(validIotKey\s*=\s*process\.env\.IOT_API_KEY \|\| )['"].*?['"]/g, '$1"[REDACTED]"');
    
    // RTSP Credentials
    content = content.replace(/rtsp:\/\/[^@]+@/g, 'rtsp://[REDACTED_USER]:[REDACTED_PASSWORD]@');
    
    // Passwords / DB Credentials
    content = content.replace(/(POSTGRES_PASSWORD:\s*).*/g, '$1[REDACTED]');
    content = content.replace(/(MINIO_ROOT_PASSWORD:\s*).*/g, '$1[REDACTED]');
    content = content.replace(/(DATABASE_URL=).*/g, '$1postgresql://[USER]:[PASSWORD]@[HOST]:[PORT]/[DB]?schema=public');
    
    return content;
}

function copyFile(src, dst) {
    if (!fs.existsSync(src)) return false;
    fs.mkdirSync(path.dirname(dst), { recursive: true });
    
    try {
        const ext = path.extname(src).toLowerCase();
        const binaryExts = ['.pth', '.onnx', '.h5', '.png', '.jpg', '.jpeg', '.zip', '.tar', '.gz'];
        if (binaryExts.includes(ext)) {
            // Exclude heavy binaries
            if (ext !== '.pth' && ext !== '.onnx' && ext !== '.h5') {
                 fs.copyFileSync(src, dst);
            }
            return true;
        }
        let content = fs.readFileSync(src, 'utf-8');
        let redacted = redactContent(content);
        fs.writeFileSync(dst, redacted, 'utf-8');
    } catch (e) {
        console.log(`Skipping file: ${src} (${e.message})`);
        return false;
    }
    return true;
}

function copyDir(src, dst, includeExts = null, excludeDirs = null) {
    if (!fs.existsSync(src)) return false;
    excludeDirs = excludeDirs || ['.git', 'node_modules', '.next', '__pycache__', 'venv', 'models'];
    
    let copied = false;
    const items = fs.readdirSync(src, { withFileTypes: true });
    for (const item of items) {
        const srcPath = path.join(src, item.name);
        const dstPath = path.join(dst, item.name);
        
        if (item.isDirectory()) {
            if (!excludeDirs.includes(item.name)) {
                if (copyDir(srcPath, dstPath, includeExts, excludeDirs)) {
                    copied = true;
                }
            }
        } else {
            if (includeExts && !includeExts.some(ext => item.name.endsWith(ext))) {
                continue;
            }
            if (copyFile(srcPath, dstPath)) {
                copied = true;
            }
        }
    }
    return copied;
}

function main() {
    if (fs.existsSync(STAGING)) {
        fs.rmSync(STAGING, { recursive: true, force: true });
    }
    fs.mkdirSync(STAGING, { recursive: true });
    
    let manifest = ["# SOURCE MANIFEST\n\n## Included Files:\n"];
    let notFound = ["\n## Not Found / Excluded Files:\n"];
    
    function addToManifest(p, success, note = "") {
        if (success) {
            manifest.push(`- \`[FOUND]\` ${p} ${note}\n`);
        } else {
            notFound.push(`- \`[NOT FOUND / EXCLUDED]\` ${p}\n`);
        }
    }

    // 1. ml-service
    const mlSrc = path.join(WORKSPACE, 'ml-service');
    const mlDst = path.join(STAGING, 'ml-service');
    addToManifest("ml-service/main.py", copyFile(path.join(mlSrc, 'main.py'), path.join(mlDst, 'main.py')));
    addToManifest("ml-service/Dockerfile", copyFile(path.join(mlSrc, 'Dockerfile'), path.join(mlDst, 'Dockerfile')));
    addToManifest("ml-service/requirements.txt", copyFile(path.join(mlSrc, 'requirements.txt'), path.join(mlDst, 'requirements.txt')));
    
    if (fs.existsSync(mlSrc)) {
        const mlItems = fs.readdirSync(mlSrc);
        for (const item of mlItems) {
            if (item.endsWith('.py') && item !== 'main.py') {
                copyFile(path.join(mlSrc, item), path.join(mlDst, item));
                addToManifest(`ml-service/${item}`, true, "(local import script)");
            }
        }
    }
    
    // 2. disguise-backend
    const bSrc = path.join(WORKSPACE, 'disguise-backend');
    const bDst = path.join(STAGING, 'disguise-backend');
    addToManifest("backend/src/queues/inference.worker.ts", copyFile(path.join(bSrc, 'src/queues', 'inference.worker.ts'), path.join(bDst, 'src/queues', 'inference.worker.ts')));
    addToManifest("backend/src/queues/index.ts", copyFile(path.join(bSrc, 'src/queues', 'index.ts'), path.join(bDst, 'src/queues', 'index.ts')));
    addToManifest("backend/src/modules/inference/", copyDir(path.join(bSrc, 'src/modules', 'inference'), path.join(bDst, 'src/modules', 'inference')));
    addToManifest("backend/src/modules/watchlist/", copyDir(path.join(bSrc, 'src/modules', 'watchlist'), path.join(bDst, 'src/modules', 'watchlist')));
    addToManifest("backend/src/sockets/", copyDir(path.join(bSrc, 'src/sockets'), path.join(bDst, 'src/sockets')));
    addToManifest("backend/src/config/minio.ts", copyFile(path.join(bSrc, 'src/config', 'minio.ts'), path.join(bDst, 'src/config', 'minio.ts')));
    addToManifest("backend/prisma/schema.prisma", copyFile(path.join(bSrc, 'prisma', 'schema.prisma'), path.join(bDst, 'prisma', 'schema.prisma')));
    addToManifest("backend/prisma/migrations/", copyDir(path.join(bSrc, 'prisma', 'migrations'), path.join(bDst, 'prisma', 'migrations')));
    addToManifest("backend/docker-compose.yml", copyFile(path.join(bSrc, 'docker-compose.yml'), path.join(bDst, 'docker-compose.yml')));
    addToManifest("backend/.env.example", copyFile(path.join(bSrc, '.env.example'), path.join(bDst, '.env.example')));
    addToManifest("backend/mediamtx.yml", copyFile(path.join(bSrc, 'mediamtx.yml'), path.join(bDst, 'mediamtx.yml')));

    // 3. disguise-raps/v2
    const rSrc = path.join(WORKSPACE, 'disguise-raps', 'v2');
    const rDst = path.join(STAGING, 'disguise-raps', 'v2');
    addToManifest("disguise-raps/v2/disguiseid_capture.py", copyFile(path.join(rSrc, 'disguiseid_capture.py'), path.join(rDst, 'disguiseid_capture.py')));
    
    if (fs.existsSync(rSrc)) {
        const rItems = fs.readdirSync(rSrc);
        for (const item of rItems) {
            if (item.endsWith('.py') && item !== 'disguiseid_capture.py') {
                copyFile(path.join(rSrc, item), path.join(rDst, item));
                addToManifest(`disguise-raps/v2/${item}`, true, "(local import script)");
            }
        }
    }

    // 4. disguise-frontend
    const fSrc = path.join(WORKSPACE, 'disguise-frontend');
    const fDst = path.join(STAGING, 'disguise-frontend');
    addToManifest("frontend/services/socket.ts", copyFile(path.join(fSrc, 'services', 'socket.ts'), path.join(fDst, 'services', 'socket.ts')));
    addToManifest("frontend/store/alertStore.ts", copyFile(path.join(fSrc, 'store', 'alertStore.ts'), path.join(fDst, 'store', 'alertStore.ts')));
    
    const alertComps = path.join(fSrc, 'components', 'alerts');
    addToManifest("frontend/components/alerts/", copyDir(alertComps, path.join(fDst, 'components', 'alerts')));
    
    const alertPage = path.join(fSrc, 'app', '(dashboard)', 'alerts');
    if (fs.existsSync(alertPage)) {
        addToManifest("frontend/app/(dashboard)/alerts/", copyDir(alertPage, path.join(fDst, 'app', '(dashboard)', 'alerts')));
    } else {
        const alertPage2 = path.join(fSrc, 'app', 'alerts');
        addToManifest("frontend/app/alerts/", copyDir(alertPage2, path.join(fDst, 'app', 'alerts')));
    }
    
    const livePage = path.join(fSrc, 'app', '(dashboard)', 'live');
    if (fs.existsSync(livePage)) {
        addToManifest("frontend/app/(dashboard)/live/", copyDir(livePage, path.join(fDst, 'app', '(dashboard)', 'live')));
    } else {
        const livePage2 = path.join(fSrc, 'app', 'live');
        addToManifest("frontend/app/live/", copyDir(livePage2, path.join(fDst, 'app', 'live')));
    }
    
    addToManifest("frontend/services/", copyDir(path.join(fSrc, 'services'), path.join(fDst, 'services')));
    
    // Write manifest
    fs.writeFileSync(path.join(STAGING, 'SOURCE_MANIFEST.md'), manifest.join('') + notFound.join(''));
    
    console.log('Creating ZIP via PowerShell Compress-Archive');
}

main();

File: extract_for_review.py
import os
import shutil
import zipfile
import re
from pathlib import Path

WORKSPACE = Path(r"f:\PROJECT\DISGUISE-ID\fullstack-disguise")
STAGING = WORKSPACE / "staging_extract"
ZIP_NAME = WORKSPACE / "disguise-ai-integration-source.zip"

def redact_content(content):
    # Redact sensitive information
    # JWT Secrets
    content = re.sub(r'(JWT_SECRET=)[\w\-\.]+', r'\1[REDACTED]', content)
    content = re.sub(r'(jwt\.verify\(.*?, process\.env\.JWT_SECRET \|\| )[''"].*?[''"]', r'\1"[REDACTED]"', content)
    
    # API Keys
    content = re.sub(r'(x-api-key)[\w\-]+', r'\1[REDACTED]', content)
    content = re.sub(r'(IOT_API_KEY\s*=\s*)[''"].*?[''"]', r'\1"[REDACTED]"', content)
    content = re.sub(r'(IOT_API_KEY\s*=\s*os\.getenv\("IOT_API_KEY",\s*)[''"].*?[''"]', r'\1"[REDACTED]"', content)
    content = re.sub(r'(validIotKey\s*=\s*process\.env\.IOT_API_KEY \|\| )[''"].*?[''"]', r'\1"[REDACTED]"', content)
    
    # RTSP Credentials
    content = re.sub(r'rtsp://[^@]+@', r'rtsp://[REDACTED_USER]:[REDACTED_PASSWORD]@', content)
    
    # Passwords / DB Credentials
    content = re.sub(r'(POSTGRES_PASSWORD: ).*', r'\1[REDACTED]', content)
    content = re.sub(r'(MINIO_ROOT_PASSWORD: ).*', r'\1[REDACTED]', content)
    content = re.sub(r'(DATABASE_URL=).*', r'\1postgresql://[USER]:[PASSWORD]@[HOST]:[PORT]/[DB]?schema=public', content)
    
    return content

def copy_file(src, dst):
    if not src.exists():
        return False
    dst.parent.mkdir(parents=True, exist_ok=True)
    
    try:
        content = src.read_text(encoding='utf-8', errors='ignore')
        redacted = redact_content(content)
        dst.write_text(redacted, encoding='utf-8')
    except Exception as e:
        print(f"Skipping binary/unreadable file: {src} ({e})")
        return False
    return True

def copy_dir(src, dst, include_exts=None, exclude_dirs=None):
    if not src.exists() or not src.is_dir():
        return False
    exclude_dirs = exclude_dirs or ['.git', 'node_modules', '.next', '__pycache__', 'venv', 'models']
    
    copied = False
    for root, dirs, files in os.walk(src):
        dirs[:] = [d for d in dirs if d not in exclude_dirs]
        for file in files:
            if include_exts and not any(file.endswith(ext) for ext in include_exts):
                continue
            s = Path(root) / file
            rel = s.relative_to(src)
            d = dst / rel
            if copy_file(s, d):
                copied = True
    return copied

def main():
    if STAGING.exists():
        shutil.rmtree(STAGING)
    STAGING.mkdir()
    
    manifest = ["# SOURCE MANIFEST\n\n## Included Files:\n"]
    not_found = ["\n## Not Found / Excluded Files:\n"]
    
    def add_to_manifest(path, success, note=""):
        if success:
            manifest.append(f"- [FOUND] {path} {note}\n")
        else:
            not_found.append(f"- [NOT FOUND / EXCLUDED] {path}\n")

    # 1. ml-service
    ml_src = WORKSPACE / 'ml-service'
    ml_dst = STAGING / 'ml-service'
    add_to_manifest("ml-service/main.py", copy_file(ml_src / 'main.py', ml_dst / 'main.py'))
    add_to_manifest("ml-service/Dockerfile", copy_file(ml_src / 'Dockerfile', ml_dst / 'Dockerfile'))
    add_to_manifest("ml-service/requirements.txt", copy_file(ml_src / 'requirements.txt', ml_dst / 'requirements.txt'))
    # Local imports (custom_loader.py, etc)
    for p in ml_src.glob('*.py'):
        if p.name not in ['main.py']:
            copy_file(p, ml_dst / p.name)
            add_to_manifest(f"ml-service/{p.name}", True, "(local import script)")
    
    # 2. disguise-backend
    b_src = WORKSPACE / 'disguise-backend'
    b_dst = STAGING / 'disguise-backend'
    add_to_manifest("backend/src/queues/inference.worker.ts", copy_file(b_src / 'src/queues/inference.worker.ts', b_dst / 'src/queues/inference.worker.ts'))
    add_to_manifest("backend/src/queues/index.ts", copy_file(b_src / 'src/queues/index.ts', b_dst / 'src/queues/index.ts'))
    add_to_manifest("backend/src/modules/inference/", copy_dir(b_src / 'src/modules/inference', b_dst / 'src/modules/inference'))
    add_to_manifest("backend/src/modules/watchlist/", copy_dir(b_src / 'src/modules/watchlist', b_dst / 'src/modules/watchlist'))
    add_to_manifest("backend/src/sockets/", copy_dir(b_src / 'src/sockets', b_dst / 'src/sockets'))
    add_to_manifest("backend/src/config/minio.ts", copy_file(b_src / 'src/config/minio.ts', b_dst / 'src/config/minio.ts'))
    add_to_manifest("backend/prisma/schema.prisma", copy_file(b_src / 'prisma/schema.prisma', b_dst / 'prisma/schema.prisma'))
    add_to_manifest("backend/prisma/migrations/", copy_dir(b_src / 'prisma/migrations', b_dst / 'prisma/migrations'))
    add_to_manifest("backend/docker-compose.yml", copy_file(b_src / 'docker-compose.yml', b_dst / 'docker-compose.yml'))
    add_to_manifest("backend/.env.example", copy_file(b_src / '.env.example', b_dst / '.env.example'))
    add_to_manifest("backend/mediamtx.yml", copy_file(b_src / 'mediamtx.yml', b_dst / 'mediamtx.yml'))

    # 3. disguise-raps/v2
    r_src = WORKSPACE / 'disguise-raps' / 'v2'
    r_dst = STAGING / 'disguise-raps' / 'v2'
    add_to_manifest("disguise-raps/v2/disguiseid_capture.py", copy_file(r_src / 'disguiseid_capture.py', r_dst / 'disguiseid_capture.py'))
    # Local imports
    for p in r_src.glob('*.py'):
        if p.name not in ['disguiseid_capture.py']:
            copy_file(p, r_dst / p.name)
            add_to_manifest(f"disguise-raps/v2/{p.name}", True, "(local import script)")

    # 4. disguise-frontend
    f_src = WORKSPACE / 'disguise-frontend'
    f_dst = STAGING / 'disguise-frontend'
    add_to_manifest("frontend/services/socket.ts", copy_file(f_src / 'services/socket.ts', f_dst / 'services/socket.ts'))
    add_to_manifest("frontend/store/alertStore.ts", copy_file(f_src / 'store/alertStore.ts', f_dst / 'store/alertStore.ts'))
    # Alert components and pages
    f_comps = f_src / 'components' / 'alerts'
    if f_comps.exists():
         add_to_manifest("frontend/components/alerts/", copy_dir(f_comps, f_dst / 'components/alerts'))
    else:
         add_to_manifest("frontend/components/alerts/", False)
         
    # search for alert/detection pages
    f_pages_alert = f_src / 'app' / '(dashboard)' / 'alerts'
    if f_pages_alert.exists():
         add_to_manifest("frontend/app/(dashboard)/alerts/", copy_dir(f_pages_alert, f_dst / 'app/(dashboard)/alerts'))
    else:
         # Try app/alerts
         f_pages_alert2 = f_src / 'app' / 'alerts'
         if f_pages_alert2.exists():
              add_to_manifest("frontend/app/alerts/", copy_dir(f_pages_alert2, f_dst / 'app/alerts'))
         else:
              add_to_manifest("frontend/alerts_page", False)
              
    f_pages_detection = f_src / 'app' / '(dashboard)' / 'live'
    if f_pages_detection.exists():
         add_to_manifest("frontend/app/(dashboard)/live/", copy_dir(f_pages_detection, f_dst / 'app/(dashboard)/live'))
    else:
         f_pages_detection2 = f_src / 'app' / 'live'
         if f_pages_detection2.exists():
              add_to_manifest("frontend/app/live/", copy_dir(f_pages_detection2, f_dst / 'app/live'))
         else:
             add_to_manifest("frontend/live_page", False)
             
    f_services = f_src / 'services'
    if f_services.exists():
         add_to_manifest("frontend/services/", copy_dir(f_services, f_dst / 'services'))
         
    # Generate Manifest
    manifest_content = ''.join(manifest) + ''.join(not_found)
    (STAGING / 'SOURCE_MANIFEST.md').write_text(manifest_content)
    
    # Zip
    print(f'Creating zip {ZIP_NAME}')
    with zipfile.ZipFile(ZIP_NAME, 'w', zipfile.ZIP_DEFLATED) as zipf:
        for root, _, files in os.walk(STAGING):
            for file in files:
                file_path = os.path.join(root, file)
                arcname = os.path.relpath(file_path, STAGING)
                zipf.write(file_path, arcname)
    
    print('Done!')

if __name__ == '__main__':
    main()
